src/pages/task/taskTabs/ExecutionTasksTable.tsx

```diff
@@ -1,69 +1,106 @@
+import { TableQueryParams, onChangeHandler } from "components/Table/utils";
+import { getColumnsTemplate } from "components/TasksTable/Columns";
+import { TaskTableInfo } from "components/TasksTable/types";
+import {
+  TaskQuery,
+  TaskSortCategory,
+  SortDirection,
+} from "gql/generated/types";
+import { useTableSort } from "hooks";
+import { useQueryParam } from "hooks/useQueryParam";
 import { useUpdateURLQueryParams } from "hooks/useUpdateURLQueryParams";
-import { RequiredQueryParams, TableOnChange } from "types/task";
-import { queryString } from "utils";
 
-const { parseQueryString, parseSortString, toSortString } = queryString;
+const { getDefaultOptions: getDefaultSorting } = Sorting;
 
 interface Props {
   execution: number;
   executionTasksFull: TaskQuery["task"]["executionTasksFull"];
   isPatch: boolean;
 }
 
-const useSorts = (): SortOrder[] => {
-  const location = useLocation();
-
-  const { sorts } = parseQueryString(location.search);
-  return parseSortString(sorts);
-};
-
 export const ExecutionTasksTable: React.FC<Props> = ({
   execution,
   executionTasksFull,
   isPatch,
 }) => {
-  const taskAnalytics = useTaskAnalytics();
+  const { sendEvent } = useTaskAnalytics();
   const updateQueryParams = useUpdateURLQueryParams();
-  const sorts = useSorts();
+
+  const [sortBy] = useQueryParam(TableQueryParams.SortBy, "");
+  const [sortDir] = useQueryParam(TableQueryParams.SortDir, "");
+
+  // Apply default sort if none is defined.
   useEffect(() => {
-    if (sorts.length === 0) {
+    if (!sortBy && !sortDir) {
       updateQueryParams({
-        sorts: "STATUS:ASC",
+        sortBy: TaskSortCategory.Status,
+        sortDir: SortDirection.Asc,
       });
     }
   }, []); // eslint-disable-line react-hooks/exhaustive-deps
 
-  const tableChangeHandler: TableOnChange<Task> = (...[, , sorter]) => {
-    updateQueryParams({
-      sorts: toSortString(sorter),
-      [RequiredQueryParams.Execution]: `${execution}`,
-    });
-  };
   const uniqueExecutions = new Set([
     execution,
     ...executionTasksFull.map((t) => t.execution),
   ]);
+
+  const columns = useMemo(
+    () =>
+      getColumnsTemplate({
+        isPatch,
+        onClickTaskLink: () => sendEvent({ name: "Click Execution Task Link" }),
+        showTaskExecutionLabel: uniqueExecutions.size > 1,
+      }),
+    [isPatch, sendEvent, uniqueExecutions.size],
+  );
+
+  const tableSortHandler = useTableSort({
+    sendAnalyticsEvents: (sorter: SortingState) =>
+      sendEvent({
+        name: "Sort Execution Tasks Table",
+        sortBy: sorter.map(({ id }) => id as TaskSortCategory),
+      }),
+  });
+
+  const tableContainerRef = useRef<HTMLDivElement>(null);
+  const table: LeafyGreenTable<TaskTableInfo> =
+    useLeafyGreenTable<TaskTableInfo>({
+      columns,
+      containerRef: tableContainerRef,
+      data: executionTasksFull ?? [],
+      defaultColumn: {
+        enableColumnFilter: false,
```

Comment:
The current implementation allows multisort but it's missing here. You can check out the tests table for an example!

Reply:
Ok I just noticed that I don't think this works currently 🤣 do you think we should just omit multisort altogether?

Reply:
added it back! i think it works now...